File: challenge_2/python/returnlove/src/single_number.py
"""
To find the single number in a list.
	Counts number of times each element is present in the list
	prints all the elements which are present only once
"""

# function takes input list and converts to a dictionary, key being the element and value is the count
def single_number(input_list):
	my_dict = {}
	for element in input_list:
		if element in my_dict.keys():
			my_dict[element] += 1
		else:
			my_dict[element] = 1
	# find the keys in dict where the count is one
	for k in my_dict.keys():
		if my_dict[k] == 1:
			print(k)

def main():
	my_list = [1,2,2,3,3,4,4,5,5,6,6,'x','x','y']
	single_number(my_list)


if __name__ == "__main__":
	main()

File: challenge_2/swift/stevenr12/single_number.swift
var array : [Int] = [2,3,4,2,3,5,4,6,4,6,9,10,9,8,7,8,10,7]
var counts : [Int:Int] = [:]
for value in array{
    counts[value] = (counts[value] ?? 0) + 1
}
print(counts.filter{$0.value == 1}[0].key)